build.gradle

```diff
@@ -268,7 +268,7 @@ subprojects {
 // the dependency is added.
 gradle.projectsEvaluated {
   allprojects {
-    if (project.path == ':test:framework') {
+    if (project.path == ':test:framework' || project.path == ':client:test') {
       // :test:framework:test cannot run before and after :core:test
```

Comment:
I don't understand your statement @nik9000. The task graph cannot have cycles, and a dependency on yourself is still a cycle.